src/generic/domain.jl

```diff
@@ -1,44 +1,46 @@
 # Definition of the abstract Domain type and its interface
 
-# The type Domain{T} is defined in IntervalSets.jl
+Domain(d) = convert(Domain, d)
 
-eltype(::Type{<:Domain{T}}) where {T} = T
-prectype(::Type{<:Domain{T}}) where {T} = prectype(T)
-numtype(::Type{<:Domain{T}}) where {T} = numtype(T)
+prectype(::Type{<:Domain{T}}) where T = prectype(T)
+numtype(::Type{<:Domain{T}}) where T = numtype(T)
 
-convert_numtype(d::Domain{T}, ::Type{U}) where {T,U} = convert(Domain{to_numtype(T,U)}, d)
-convert_prectype(d::Domain{T}, ::Type{U}) where {T,U} = convert(Domain{to_prectype(T,U)}, d)
+# Domain-specific prectype and numtype default to using domaineltype
+domain_prectype(d) = prectype(domaineltype(d))
+domain_numtype(d) = numtype(domaineltype(d))
+prectype(d::AsDomain) = prectype(domaineltype(d))
+numtype(d::AsDomain) = numtype(domaineltype(d))
 
-Domain(d) = convert(Domain, d)
+# convenience abbreviations for internal usage
+const deltype = domaineltype
+const dprectype = domain_prectype
+const dnumtype = domain_numtype
```

Comment:
I think these should be removed and find-and-replaced all occurrences: adding synonyms just adds confusion without much benefit.

Reply:
Sure, valid point